Tests/StringsLintFrameworkTests/Parsers/StructuredPlaceholderCommentParserTests.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mark Hall on 2022-07-19.
+//
+
+import XCTest
+@testable import StringsLintFramework
+
+class StructuredPlaceholderCommentParserTests: ParserTestCase {
+
+    func testParseFullFile() throws {
+
+        let content = """
+// *** This file is generated, do not make changes to it. Please see our documentation for more information on our localization process: https://faire.link/l10n ***
+
+/*
+  Retailer.strings
+  Retailer
+
+  Created by Raissa Nucci on 07/04/20.
+  Copyright © 2020 Faire Inc. All rights reserved.
+*/
+
+//--START CONTENT--
```

Comment:
why do we need this?

Reply:
The way I've written the parser, it splits the file around `;`. So the file header is included in the "first" string. The `comment` regex looks for anything surrounded by `/* */` so the header comment will get parsed out of the string and cause an error when we try to convert it to JSON. I added this in here so that I can ignore the header in the file when parsing.

Reply:
couldn't we do a file pre-processing and remove the first comment instead? I think we did something like that in our localization tool 

https://github.com/Faire/ios/blob/36609f081c4a853c3325551b2992b12ae93bc1f0/Apps/Localization/Sources/LocalizationCore/Utilities/StringsUtils.swift#L61

Reply:
yup, that's probably a better solution 😃